### PR TITLE
ZINB and NB torch distributions

### LINKED ISSUE #582: Create pytorch distributions for gamma and NB/ZINB using our parameterizations
This will clean up a lot of sampling code.

### PR DESCRIPTION
**Pull request by PierreBoyeau**
Fixes #582

### CODE REVIEW

File: scvi/inference/posterior.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
     compute_marginal_log_likelihood_scvi,
     compute_marginal_log_likelihood_autozi,
 )
-
+from scvi.models.distributions import NB, ZINB
```
Comment (adamgayoso) [inline scvi/inference/posterior.py]: Should we follow the pytorch naming conventions (e.g., NegativeBinomial?)

File: scvi/models/distributions.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    broadcast_all,
+    probs_to_logits,
+    lazy_property,
+    logits_to_probs,
+)
+
+from scvi.models.log_likelihood import log_nb_positive, log_zinb_positive
+
+
+def from_nb_params2_to_1(mu, theta, eps=1e-6):
```
Comment (adamgayoso) [inline scvi/models/distributions.py]: is there some way we can give these functions a more specific name? Also should we make them "private"? "_from..."

File: scvi/models/__init__.py [hunk 3]
```diff
@@ -15,4 +16,6 @@
     "Classifier",
     "AutoZIVAE",
     "TOTALVI",
+    "ZINB",
```
Comment (adamgayoso) [inline scvi/models/__init__.py]: I'm not sure putting this here is necessary is it?, all your imports are from scvi.models.distributions

File: scvi/models/distributions.py (hunk 2 above)
Comment (PierreBoyeau) [reply scvi/models/distributions.py]: tried my best but it's not that easy

**Pull request merged**